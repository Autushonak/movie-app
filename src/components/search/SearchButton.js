import styled from 'styled-components';
import { colors } from '../../constants/colors';

const SearchButton = styled.button`
  flex-grow: 0.5;
  background-color: ${colors.PINK};
  border: none;
  border-radius: 4px;
  text-transform: uppercase;
  color: ${colors.WHITE};
  padding: 15px 60px;
  text-align: center;
  font-size: 16px;
`;

export default SearchButton;
