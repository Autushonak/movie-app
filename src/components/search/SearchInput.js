import React from 'react';
import styled from 'styled-components';
import { colors } from '../../constants/colors';

const SearchInputStyled = styled.input`
  flex-grow: 10;
  background-color: ${colors.GREY_MAIN};
  color: ${colors.WHITE};
  width: auto;
  opacity: 0.5;
  font-size: 16px;
  padding: 15px 5px;
  border: none;
  border-radius: 4px;
  margin-right: 10px;
`;

const SearchInput = () => <SearchInputStyled />;

export default SearchInput;
