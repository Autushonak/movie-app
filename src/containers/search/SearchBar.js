import React, { useState, useCallback } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { SearchButton, SearchInput } from '../../components';
import { getMovies, changeSearch } from '../../store/actions/movie';

const SearchBarStyled = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
  margin-top: 15px;
`;

const SearchBar = ({ getMovies, changeSearch }) => {
  const [search, setSearch] = useState('');
  const handleSearch = useCallback(() => {
    changeSearch(search);
    getMovies();
  }, [search]);
  const onChange = useCallback((e) => {
    setSearch(e.target.value);
  }, []);
  return (
    <SearchBarStyled>
      <SearchInput onChange={onChange} value={search} />
      <SearchButton onClick={handleSearch}>Search</SearchButton>
    </SearchBarStyled>
  );
};

SearchBar.propTypes = {
  getMovies: PropTypes.func,
  changeSearch: PropTypes.func,
};

const mapDispatchToProps = {
  getMovies,
  changeSearch,
};

export default connect(null, mapDispatchToProps)(SearchBar);
export { SearchBarStyled, SearchBar };
