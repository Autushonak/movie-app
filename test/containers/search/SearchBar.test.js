import 'jest-styled-components';
import React from 'react';
import { shallow } from 'enzyme';
import { SearchBarStyled, SearchBar } from '../../../src/containers/search/SearchBar';

describe('SearchBarStyled', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<SearchBarStyled />);
    expect(component).toMatchSnapshot();
  });
});

describe('SearchBar', () => {
  test('should be rendered correctly', () => {
    const component = shallow(<SearchBar />);
    expect(component).toMatchSnapshot();
  });
});
