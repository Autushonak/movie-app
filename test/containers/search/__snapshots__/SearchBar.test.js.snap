// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should be rendered correctly 1`] = `
<styled.div>
  <styled.input
    onChange={[Function]}
    value=""
  />
  <styled.button
    onClick={[Function]}
  >
    Search
  </styled.button>
</styled.div>
`;

exports[`SearchBarStyled should be rendered correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  margin-top: 15px;
}

<div
  className="c0"
/>
`;
